File: FieldLog/FieldLog/Models/SOTASummit.swift
import Foundation
import GRDB

struct SOTASummit: Codable, Identifiable, Equatable {
    var code: String              // e.g. "W4C/CM-001"
    var codeNormalized: String?   // e.g. "W4CCM001"
    var name: String
    var associationCode: String?
    var regionCode: String?
    var altitude: Int?            // Meters
    var points: Int?
    var grid: String?
    var latitude: Double?
    var longitude: Double?
    var validFrom: String?
    var validTo: String?

    var id: String { code }

    /// Display string: "W4C/CM-001 Mount Mitchell"
    var displayName: String {
        "\(code) \(name)"
    }

    /// Creates a normalized code by stripping slashes and dashes
    static func normalize(_ code: String) -> String {
        code.replacingOccurrences(of: "/", with: "")
            .replacingOccurrences(of: "-", with: "")
            .uppercased()
    }
}

extension SOTASummit: FetchableRecord, PersistableRecord {
    static var databaseTableName = "sotaSummit"
}

File: android/app/src/main/kotlin/com/sotalog/android/domain/models/SOTASummit.kt
package com.sotalog.android.domain.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "sotaSummit")
data class SOTASummit(
    @PrimaryKey
    val code: String,
    val codeNormalized: String? = null,
    val name: String,
    val associationCode: String? = null,
    val regionCode: String? = null,
    val altitude: Int? = null,
    val points: Int? = null,
    val grid: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val validFrom: String? = null,
    val validTo: String? = null,
) {
    val displayName: String get() = "$code $name"

    companion object {
        fun normalize(code: String): String =
            code.replace("/", "").replace("-", "").uppercase()
    }
}
